enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion.tsx

```diff
@@ -102,3 +98,9 @@ export const InteractiveQuestion = ({
     </LoadingAndErrorWrapper>
   );
 };
+
+export const InteractiveQuestion = (props: InteractiveQuestionProps) => (
```

Comment:
Since we'll do it in multiple components, maybe a HOC?
```diff
diff --git a/enterprise/frontend/src/embedding-sdk/components/private/PublicComponentWrapper.tsx b/enterprise/frontend/src/embedding-sdk/components/private/PublicComponentWrapper.tsx
index cd4071c939..65d52eeba5 100644
--- a/enterprise/frontend/src/embedding-sdk/components/private/PublicComponentWrapper.tsx
+++ b/enterprise/frontend/src/embedding-sdk/components/private/PublicComponentWrapper.tsx
@@ -1,3 +1,4 @@
+import type { ComponentType } from "react";
 import { t } from "ttag";
 
 import { SdkError } from "embedding-sdk/components/private/SdkError";
@@ -29,3 +30,21 @@ export const PublicComponentWrapper = ({
 
   return children;
 };
+
+export function withPublicComponentWrapper<P>(
+  WrappedComponent: ComponentType<P>,
+): React.FC<P> {
+  const WithPublicComponentWrapper: React.FC<P> = props => {
+    return (
+      <PublicComponentWrapper>
+        <WrappedComponent {...props} />
+      </PublicComponentWrapper>
+    );
+  };
+
+  WithPublicComponentWrapper.displayName = `withPublicComponentWrapper(${
+    WrappedComponent.displayName || WrappedComponent.name || "Component"
+  })`;
+
+  return WithPublicComponentWrapper;
+}
diff --git a/enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion.tsx b/enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion.tsx
index 6fa02a61dd..a086d100bd 100644
--- a/enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion.tsx
+++ b/enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion.tsx
@@ -1,7 +1,7 @@
 import cx from "classnames";
 import { useEffect } from "react";
 
-import { PublicComponentWrapper } from "embedding-sdk/components/private/PublicComponentWrapper";
+import { withPublicComponentWrapper } from "embedding-sdk/components/private/PublicComponentWrapper";
 import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
 import CS from "metabase/css/core/index.css";
 import { useDispatch, useSelector } from "metabase/lib/redux";
@@ -99,8 +99,5 @@ export const _InteractiveQuestion = ({
   );
 };
 
-export const InteractiveQuestion = (props: InteractiveQuestionProps) => (
-  <PublicComponentWrapper>
-    <_InteractiveQuestion {...props} />
-  </PublicComponentWrapper>
-);
+export const InteractiveQuestion =
+  withPublicComponentWrapper(_InteractiveQuestion);
```

Reply:
Yeah, I thought about this as well but I know we're trying to move away from HoCs in general (i.e. in `DashboardControls` below for example) 

<img width="617" alt="image" src="https://github.com/metabase/metabase/assets/25306947/5660af15-5cd9-4099-8fdf-f1e30bcc0e9b">

Reply:
I think HOCs are generally deprecated for adding logic and props as hooks provide a better alternative, but hooks don't really provide an alternative for "wrap and conditionally render". We would have to use `const flag = useCustomHook` and then `if(flag) return null`.
Between manual wrapping and the HOC I think HOCs look better, but let's see what the team thinks

Reply:
Yeah, I'm not totally tied to following the rules so I changed it to an HoC. let me know what you think